### PR TITLE
Azure mqtt publisher

### PR DESCRIPTION
**Pull request by alenmagdic**
**Description**
Implemented publishing advertisement data to Azure IOT Hub. Implemented creating, updating and deleting of Azure MQTT publishers. Implemented testing Azure MQTT publisher.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -66,6 +70,8 @@ dependencies {
     implementation "androidx.constraintlayout:constraintlayout-solver:$constraint_layout_version"
     implementation "androidx.lifecycle:lifecycle-extensions:2.0.0"
     implementation "androidx.room:room-runtime:2.1.0"
+    implementation 'androidx.appcompat:appcompat:1.0.2'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
```
Comment (borna-mdn) [inline app/build.gradle]: Redundant! These 2 dependencies are already declared.

File: app/build.gradle [hunk 2]
```diff
@@ -50,6 +50,10 @@ android {
             dimension "version"
         }
     }
+
+    packagingOptions {
+        exclude 'thirdpartynotice.txt'
+    }
```
Comment (borna-mdn) [inline app/build.gradle]: Why do we need this?

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -33,6 +33,8 @@
         android:theme="@style/AppTheme"
         android:usesCleartextTraffic="true"
         tools:ignore="GoogleAppIndexingWarning">
+        <activity android:name=".ui.settings.publishers.selectpublish.AzureMqttPublisherActivity"></activity>
+        <activity android:name=".ui.MainActivity" />
```
Comment (borna-mdn) [inline app/src/main/AndroidManifest.xml]: MainActivity entry is duplicated.

File: app/src/main/java/com/aconno/sensorics/BluetoothScanningService.kt [hunk 4]
```diff
@@ -261,7 +269,8 @@ class BluetoothScanningService : DaggerService() {
                 Observable.merge(
                     getGooglePublisherObservable(),
                     getRestPublisherObservable(),
-                    getMqttPublishers()
+                    getMqttPublishers(),
+                        getAzureMqttPublishers()
```
Comment (borna-mdn) [inline app/src/main/java/com/aconno/sensorics/BluetoothScanningService.kt]: Fix this formatting.

File: app/build.gradle (hunk 2 above)
Comment (alenmagdic) [reply app/build.gradle]: Because Azure IOT Sdk doesn't work if we don't exclude this file

**Pull request merged**